### PR TITLE
Add _GERMLINE and _DRIVER support to oncoprinter

### PR DESCRIPTION
**Pull request by adamabeshouse**
Adds support for users of oncoprinter to specify mutations as germlines and drivers.

image

### CODE REVIEW

File: src/pages/staticPages/tools/oncoprinter/Oncoprinter.tsx [hunk 1]
```diff
@@ -324,6 +353,18 @@ export default class Oncoprinter extends React.Component<IOncoprinterProps, {}>
                      onMouseEnter={this.onMouseEnter}
                      onMouseLeave={this.onMouseLeave}
                 >
+                    { this.props.store.existCustomDrivers && !this.props.store.customDriverWarningHidden &&
+                        this.props.store.driverAnnotationSettings.customBinary && (
```
Comment (alisman) [inline src/pages/staticPages/tools/oncoprinter/Oncoprinter.tsx]: what does customBinary mean?

File: src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts [hunk 2]
```diff
@@ -228,6 +262,9 @@ export function makeGeneticTrackDatum_Data_Type2(oncoprinterInputLine:Oncoprinte
         // these are the same always or almost always
         hugoGeneSymbol:oncoprinterInputLine.hugoGeneSymbol,
         proteinChange:oncoprinterInputLine.proteinChange,
+        mutationStatus: oncoprinterInputLine.isGermline ? "germline": "",
```
Comment (alisman) [inline src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts]: seems likely there might already be constants somewhere for these (germline and Putative_Driver)

File: src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts [hunk 3]
```diff
@@ -509,7 +551,12 @@ export function annotateGeneticTrackData(
                 }
                 if (d.molecularProfileAlterationType === AlterationTypeConstants.MUTATION_EXTENDED) {
                     // tag mutations as putative driver, and filter them
-                    d.putativeDriver = !!(d.oncoKbOncogenic || (params.useHotspots && d.isHotspot) || getCBioAnnotation(d));
+                    d.putativeDriver = !!(
+                        d.oncoKbOncogenic ||
```
Comment (alisman) [inline src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts]: is this covered in test?  seems like a complicated condition that would qualify as a "unit" for test

File: src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts [hunk 4]
```diff
@@ -567,17 +614,46 @@ export function parseGeneticInput(input:string):{status:"complete", result:Oncop
                             throw new Error(`${errorPrefix}Alteration "${alteration}" is not valid - it must be "HIGH" or "LOW" if Type is "PROT"`);
                         }
                         break;
+                    case "fusion":
```
Comment (alisman) [inline src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts]: is handling fusion part of scope of PR?

File: src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts [hunk 5]
```diff
@@ -567,17 +614,46 @@ export function parseGeneticInput(input:string):{status:"complete", result:Oncop
...
+                        for (const modifier of parsedMutation.modifiers) {
+                            switch (modifier) {
+                                case "GERMLINE":
+                                    ret.isGermline = true;
+                                    break;
+                                case "DRIVER":
+                                    ret.isCustomDriver = true;
+                                    break;
+                                default:
+                                    throw new Error(`${errorPrefix}Only allowed mutation modifiers are GERMLINE and DRIVER`);
```
Comment (alisman) [inline src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts]: it's nice that you are throwing these errors.  we should discuss how we are presenting these kind of errors to user.  i know we discussed error boundaries before.  i think you found that they are not compatible with mobx.  lets discuss

File: src/pages/staticPages/tools/oncoprinter/Oncoprinter.tsx (hunk 1 above)
Comment (adamabeshouse) [reply src/pages/staticPages/tools/oncoprinter/Oncoprinter.tsx]: thats the name we've had for custom binary driver annotations (binary meaning -driver yes, or driver no. as opposed to custom *tiered* annotations)

File: src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts (hunk 4 above)
Comment (adamabeshouse) [reply src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts]: no, I just have to handle it separately now that I am parsing mutations differently due to the _GERMLINE and _DRIVER, whereas before I was handling it together with other mutations.

File: src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts (hunk 3 above)
Comment (adamabeshouse) [reply src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts]: ehhhh... its realy just an OR statement, it looks a little ugly but its logically just saying that its a putative driver if any of the selected sources give it a driver annotation

Comment (adamabeshouse) [reply src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts]: its implicitly covered by the e2e tests about annotation

File: src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts (hunk 5 above)
Comment (adamabeshouse) [reply src/pages/staticPages/tools/oncoprinter/OncoprinterGeneticUtils.ts]: I don't see the need for an error boundary here, but we can discuss

**Pull request merged**